frontend/js/src/artist-similarity/src/Data.tsx

```diff
@@ -0,0 +1,150 @@
+    const LINK_DIST_MULTIPLIER = 200;
+    const MIN_LINK_DIST = 50;
+    const MAIN_NODE_SIZE = 150;
+    const SIMILAR_NODE_SIZE = 85;
+    const BACKGROUND_ALPHA = 0.2;
+    const COLOR_MIX_WEIGHT = 0.3;
+    
+    var color1 = colorGenerator();
+    var color2 = colorGenerator();
+
+    const [similarArtists, setSimilarArtists] = useState<Array<ArtistType>>();
+    const [mainArtist, setMainArtist] = useState<ArtistType>();
+    const [similarArtistsLimit, setSimilarArtistsLimit] = useState(SIMILAR_ARTISTS_LIMIT_VALUE);
+    const [colors, setColors] = useState([color1, color2]);
+    const [artistMBID, setArtistMBID] = useState(ARTIST_MBID);
+
+    var scoreList: Array<number> = [];
+    
+    const fetchData = async (ARTIST_MBID: string): Promise<void> => {
+        try {
+          const response = await fetch(BASE_URL + ARTIST_MBID);
+          const data = await response.json();
+          console.log(data);
+          processData(data);
+        }
+        catch (error){
+          //Error message goes here.
+          alert("Something went wrong while loading information, please try again");
+        }
+    }
+    
+    const processData = (dataResponse: ApiResponseType): void => {
+        let mainArtistResponse = dataResponse[1];
+        setMainArtist(mainArtistResponse.type === "dataset" ? mainArtistResponse.data[0] : undefined);
+        
+        let similarArtistsResponce = dataResponse[3];
+        setSimilarArtists(similarArtistsResponce.type === "dataset" ? similarArtistsResponce.data.slice(0, similarArtistsLimit) : undefined);
+
+        setColors([tinycolor.mix(color1, color2, COLOR_MIX_WEIGHT), color2]);
+    }
+    
+    useEffect(() => {
+        fetchData(artistMBID);
+    }, [artistMBID]);
+
+    // Calculating minScore for normalization which is always the last element of the array (because it's sorted)
+    var minScore = similarArtists?.[similarArtistsLimit - 1]?.score ?? 0;
+    minScore = Math.sqrt(minScore);
+
+    var transformedArtists: GraphDataType = similarArtists! && {
+        
+        nodes: [mainArtist! , ...similarArtists].map((similarArtist: ArtistType, index: number): NodeType => {
+            let computedScore;
+            let computedColor;
+            if(similarArtist !== mainArtist) {
+                computedScore = minScore / Math.sqrt(similarArtist?.score ?? 1);
+                computedColor = tinycolor.mix(colors[0], colors[1], (index / SIMILAR_ARTISTS_LIMIT_VALUE * computedScore) * 100);
+                scoreList.push(computedScore);
+            }
+            
+            else {
+                computedColor = colors[0];
+            }
+
+            return {
+                id: similarArtist.name,
+                artist_mbid: similarArtist.artist_mbid,
+                size: similarArtist.artist_mbid === mainArtist?.artist_mbid ? MAIN_NODE_SIZE : SIMILAR_NODE_SIZE,
+                color: computedColor.toRgbString(),
+                score: similarArtist.score ?? 0
```

Comment:
This is indeed a good way to have a default value if score is undefined. :+1:

Just wanted to check that it is expected that the default value is `0` on this line, but it is `1` on line 112 when calculating `computedScore`

Reply:
Thinking about if again. I have redefined it as a constant
```
const NULL_SCORE = Infinity;
```
As otherwise it might interfere in case the score is really 1. Because I set up the graph to not render a score of 1 as it's the null case. Moreover, it also covers the case of division by 0.
```
computedScore = minScore / Math.sqrt(similarArtist?.score ?? NULL_SCORE);
``` 
Hope it makes sense.